pkg/controller/direct/sql/mapping.go

```diff
@@ -793,10 +800,44 @@ func Convert_SQLInstance_API_v1_To_KRM_status(in *api.DatabaseInstance, out *krm
 	return nil
 }
 
-func LazyPtr[T comparable](v T) *T {
-	var defaultValue T
-	if v == defaultValue {
-		return nil
+func SQLInstanceKRMToGCPCloneRequest(in *krm.SQLInstance, refs *SQLInstanceInternalRefs) (*api.InstancesCloneRequest, error) {
+	if in == nil {
+		return nil, fmt.Errorf("cannot convert nil SQLInstance")
+	}
+
+	if in.Spec.CloneSource == nil {
+		// spec.cloneSource is required for converting KRM.SQLInstance -> api.InstancesCloneRequest.
+		return nil, fmt.Errorf("cannot convert nil CloneSource")
+	}
+
+	cloneReq := &api.InstancesCloneRequest{
+		CloneContext: &api.CloneContext{
+			DatabaseNames: refs.sourceSQLDatabases,
+			Kind:          "sql#cloneContext",
+			PointInTime:   direct.ValueOf(in.Spec.CloneSource.PointInTime),
+		},
+	}
+
+	resourceID := ValueOf(in.Spec.ResourceID)
+	if resourceID == "" {
+		resourceID = in.Name
 	}
-	return &v
+	cloneReq.CloneContext.DestinationInstanceName = resourceID
+
+	if in.Spec.Settings.IpConfiguration != nil {
```

Comment:
Will be interesting to see if we can get this under fuzz-testing, so we can detect new fields...

Reply:
Where is the fuzz testing code? I am still just trying to achieve test coverage of the basic functionality lol.